### PR TITLE
Feature/js add smtp support

### PR DESCRIPTION
**Pull request by thatnerdjosh**
Solves #12938 

Notes:
  - Email from address should be configured in admin interface (I assume this is a non-issue as this seems to be necessary for sendmail as well)
  - SMTP configuration must be made in order for this to work correctly (how should we handle errors, or should we trust the user reads the comments in settings?)
  - This is my first PR to the codebase so I probably did something wrong, please be critical on review :)

### CODE REVIEW

File: elgg-config/settings.example.php [hunk 1]
```diff
@@ -379,6 +379,37 @@
  */
 //$CONFIG->image_processor = 'imagick';
 
+
+/**
+ * Configure emailer transport
+ * 
+ * This setting can be used to select a different emailer transport. By default the Zend Sendmail Transport is used.
+ * Currently only 'smtp' and 'sendmail' is supported as a different configuration.
```
Comment (jeabakker) [inline elgg-config/settings.example.php]: ... 'sendmail' **are** supported

File: engine/classes/Elgg/Config.php [hunk 2]
```diff
@@ -51,6 +51,8 @@
  * @property string        $elgg_settings_file
  * @property bool          $elgg_config_set_secret
  * @property bool          $enable_profiling
+ * @property string        $emailer_transport
```
Comment (jeabakker) [inline engine/classes/Elgg/Config.php]: please add a description, for future reference

File: engine/classes/Elgg/Config.php [hunk 3]
```diff
@@ -51,6 +51,8 @@
...
+ * @property array         $emailer_smtp_settings
```
Comment (jeabakker) [inline engine/classes/Elgg/Config.php]: please add a description, for future reference

**Review by jeabakker — CHANGES_REQUESTED** [review]
Just a few minor remarks.
Also have a look at http://learn.elgg.org/en/stable/contribute/code.html#commit-message-format for the correct format of the commit message.
Something like `feat(mail): add smtp support`

**Comment by jeabakker** [comment]
I thought of some more remarks
- since this is a feature it needs to be on the `3.x` branch. 
You'll need to rebase your working branch
- if possible squash all the commits into a single one

**Comment by jdalsem** [comment]
Is this tested with a working SMTP server? Or is it an assumption that this solutions should work?

**Comment by NerdsvilleCEO** [comment]
@jdalsem yes it's tested on an actual SMTP server. I tested with mailgun (try to avoid making PR's without testing :))

If someone other than me wants to volunteer to test it that'd be great to make sure my settings are understandable etc and things work for them.

Working on the changes now, apologies for the delay I'm in the hospital taking care of family.

**Review by jeabakker — APPROVED** [review]

**Comment by jeabakker** [comment]
The only problem is the format of the commit message, but we can fix that during merge

**Comment by jdalsem** [comment]
Thanks man!

**Pull request merged**